src/hooks/useKeyboardNavigation.ts

```diff
@@ -0,0 +1,313 @@
+    const currentEmojiGroup = getCurrentEmojiListGroup();
+    const nextEmojiGroup = currentEmojiGroup?.nextSibling as HTMLElement;
+
+    if (nextEmojiGroup) {
+      focusElement(nextEmojiGroup.children[columnIndex]);
+
+      const categoryName = getEmojiGroupName(nextEmojiGroup);
+      setCategory(categoryName);
+    }
+  };
+
+  const focusPrevEmojiListGroupOnLastItem = () => {
+    const currentEmojiGroup = getCurrentEmojiListGroup();
+    const prevEmojiGroup = currentEmojiGroup?.previousSibling;
+    focusElement(prevEmojiGroup?.lastChild);
+    return prevEmojiGroup;
+  };
+
+  const focusPrevEmojiListGroup = (
+    columnIndex: number,
+    itemsPerRow: number
+  ) => {
+    const currentEmojiGroup = getCurrentEmojiListGroup();
+    const prevEmojiGroup = currentEmojiGroup?.previousSibling as HTMLElement;
+
+    if (prevEmojiGroup) {
+      const numOfItems = prevEmojiGroup.children.length;
+
+      let nextFocusIndex;
+
+      for (let i = numOfItems - 1; i >= 0; i--) {
+        if (i % itemsPerRow === columnIndex) {
+          nextFocusIndex = i;
+          break;
+        }
+      }
+
+      const isInLastRow =
+        nextFocusIndex >= numOfItems - (numOfItems % itemsPerRow) ||
+        numOfItems % itemsPerRow === 0;
+
+      if (!isInLastRow) nextFocusIndex = numOfItems - 1;
+
+      focusElement(prevEmojiGroup.children[nextFocusIndex]);
+
+      const categoryName = getEmojiGroupName(prevEmojiGroup);
+      setCategory(categoryName);
+    }
+
+    return prevEmojiGroup;
+  };
+
+  // const exitSkinTones = withCatch(() => {
+  //   focusSearch();
+  // });
+
+  const focusSearch = withCatch(() => {
+    if (emojiSearchRef.current) {
+      focusElement(emojiSearchRef.current);
+    }
+  });
+
+  // const focusPrevSkinTone = () => {
+  //   const current = getActiveElement();
+  //   if (current) {
+  //     const prev = current.previousSibling;
+  //     focusElement(prev);
+  //   }
+  // };
+  // const focusNextSkinTone = () => {
+  //   const current = getActiveElement();
+
+  //   if (current) {
+  //     const next = current.nextSibling;
+  //     focusElement(next);
+  //   }
+  // };
+  // const focusActiveSkinTone = () => {
+  //   focusElement(getActiveSkinToneElement());
+  // };
+  //const getActiveSkinToneElement = () => {
+  //return skinToneSpreadRef.current.querySelector(`#t${activeSkinTone}`);
+  //};
+
+  const focusSkinTonePicker = withCatch(() => {
+    if (!skinToneSpreadRef.current) return;
+
+    //toggleSkinTonesSpread();
+    //focusActiveSkinTone();
+  });
+
+  const getCurrentSectionIndex = () => {
+    const activeElement = getActiveElement();
+    const currentSectionIndex = sections.findIndex(section =>
+      section.rootElement.current.contains(activeElement)
+    );
+
+    return currentSectionIndex;
+  };
+
+  const scrollEmojiListToTop = withCatch(() => {
+    requestAnimationFrame(() => {
+      emojiListRef.current.scrollTop = 0;
+    });
+  });
```

Comment:
I think you can use this that does essentially the same:
https://github.com/ealush/emoji-picker-react/blob/2022-08-17-v4/src/DomUtils/scrollTo.ts